include/libsmb2-private.h

```diff
@@ -27,6 +27,34 @@
 extern "C" {
 #endif
 
+#ifdef __linux__
+#ifndef __KERNEL__
+#include <endian.h>
+#else
+#include <asm/byteorder.h>
+#endif
+#endif
+
+#ifdef __FreeBSD__
+#include <sys/endian.h>
+#endif
+
+#if defined(__BYTE_ORDER) && __BYTE_ORDER == __LITTLE_ENDIAN
+# define R_ENDIAN_LITTLE
+#elif defined(__BYTE_ORDER) && __BYTE_ORDER == __BIG_ENDIAN
+# define R_ENDIAN_BIG
+#elif defined(_BYTE_ORDER) && _BYTE_ORDER == _LITTLE_ENDIAN
+# define R_ENDIAN_LITTLE
+#elif defined(_BYTE_ORDER) && _BYTE_ORDER == _BIG_ENDIAN
+# define R_ENDIAN_BIG
+#elif defined(__LITTLE_ENDIAN) && !defined(__BIG_ENDIAN)
+# define R_ENDIAN_LITTLE
+#elif !defined(__LITTLE_ENDIAN) && defined(__BIG_ENDIAN)
+# define R_ENDIAN_BIG
+#endif
```

Comment:
Do we need these macros?
We already have helper functions that can get/set 8/16/32/64 bit integers and are endian-safe :+1: 
Check out smb2_set_uint32() and friends in pdu.c.

Can we use those here ?